utils/sparse_mt.py

```diff
@@ -18,23 +18,34 @@ def compute_last_ref_block_end(mt: hl.MatrixTable) -> hl.Table:
     :rtype: Table
     """
     mt = mt.select_entries('END')
+
+    # Localize entries, so that they can be viewed as an array and scanned over using hl.scan.array_agg
     ht = mt._localize_entries('__entries', '__cols')
+
+    # Compute the position by using hl.scan._prev_nonnull.
+    # This was inspired by hl.experimental.densify
+    # _prev_non_null is an aggregator that keeps the previous record in memory
+    # and updates it with the given value at the row if it's not null (missing)
+    # The following code computes the following annotation for each row:
+    # 1. Keep a scan of the entries using _prev_nonnull, keeping the start (ht.locus) and end (entry.END) of each ref block  (1.1)
+    # 2. For the current row locus, record the start of the block that starts the furthest away,
+    #    that is the minimum position in the current scan for any block that overlaps the current locus (2.1)
```

Comment:
thank you for all of these -- they're so helpful!!

Reply:
Of course! And yes, this is complex code using undocumented (may I say secret?) hail aggregator, so additional comments were definitely warranted! :)